### PR TITLE
[ML] Additional outlier detection parameters

### PR DESCRIPTION
**Pull request by dimitris-athanasiou**
Adds the following parameters to `outlier_detection`:

- `compute_feature_influence` (boolean): whether to compute or not
   feature influence scores
- `outlier_fraction` (double): the proportion of the data set assumed
   to be outlying prior to running outlier detection
- `standardization_enabled` (boolean): whether to apply standardization
   to the feature values

### CODE REVIEW

**Comment by dimitris-athanasiou** [comment]
@szabosteve Could you please take a look at the docs changes of this PR?

**Comment by dimitris-athanasiou** [comment]
@dolaru Pinging as this might warrant new test scenarios for our QA

**Review by benwtrent — APPROVED** [review]
LGTM!

**Comment by dimitris-athanasiou** [comment]
This depends on https://github.com/elastic/ml-cpp/pull/716 to be merged first.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/OutlierDetection.java [hunk 1]
```diff
@@ -71,22 +100,31 @@ public OutlierDetection(@Nullable Integer nNeighbors, @Nullable Method method, @
...
+        this.computeFeatureInfluence = computeFeatureInfluence;
+        this.outlierFraction = outlierFraction;
+        this.standardizationEnabled = standardizationEnabled;
     }
 
     public OutlierDetection(StreamInput in) throws IOException {
         nNeighbors = in.readOptionalVInt();
         method = in.readBoolean() ? in.readEnum(Method.class) : null;
         featureInfluenceThreshold = in.readOptionalDouble();
+        if (in.getVersion().onOrAfter(Version.V_7_5_0)) {
```
[the remark this answers is not in the document]
Comment (dimitris-athanasiou) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/OutlierDetection.java]: Yes, that's why I'm holding on before merging the BWC tests :-)

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/OutlierDetectionTests.java [hunk 2]
```diff
@@ -41,20 +48,38 @@ public static OutlierDetection createRandom() {
     }
 
     public void testGetParams_GivenDefaults() {
-        OutlierDetection outlierDetection = new OutlierDetection();
-        assertThat(outlierDetection.getParams().isEmpty(), is(true));
+        OutlierDetection outlierDetection = new OutlierDetection.Builder().build();
+        Map<String, Object> params = outlierDetection.getParams();
+        assertThat(params.size(), equalTo(3));
+        assertThat(params.containsKey("compute_feature_influence"), is(true));
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/dataframe/analyses/OutlierDetectionTests.java]: FYI: There are `hasKey` and `hasEntry` methods in `org.hamcrest.Matchers` which you could use here.

**Review by szabosteve — APPROVED** [review]
Thank you for taking care of this.
LGTM.

File: docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc [hunk 3]
```diff
@@ -96,6 +96,10 @@ include-tagged::{doc-tests-file}[{api}-outlier-detection-customized]
 <1> Constructing a new OutlierDetection object
 <2> The method used to perform the analysis
 <3> Number of neighbors taken into account during analysis
+<4> The min `outlier_score` required to compute feature influence
```
Comment (przemekwitek) [inline docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc]: Just curious: Could the functionality of `compute_feature_influence` setting be achieved with setting `min_outlier_score` to a *very high* number?

**Review by przemekwitek — APPROVED** [review]
LGTM

File: docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc (hunk 3 above)
Comment (dimitris-athanasiou) [reply docs/java-rest/high-level/ml/put-data-frame-analytics.asciidoc]: Indeed, one can set `feature_influence_threshold` to `1` to achieve the same as setting `compute_feature_influence` to `false`.

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/1

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/2

**Comment by dimitris-athanasiou** [comment]
run elasticsearch-ci/1

**Pull request merged**